src/hotspot/share/utilities/rbTree.inline.hpp

```diff
@@ -0,0 +1,525 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_UTILITIES_RBTREE_INLINE_HPP
+#define SHARE_UTILITIES_RBTREE_INLINE_HPP
+
+#include "utilities/debug.hpp"
+#include "utilities/powerOfTwo.hpp"
+#include "utilities/rbTree.hpp"
+
+template <typename K, typename V, typename COMPARATOR, typename ALLOCATOR>
+inline void RBTree<K, V, COMPARATOR, ALLOCATOR>::RBNode::replace_child(
+    RBNode* old_child, RBNode* new_child) {
+  if (_left == old_child) {
+    _left = new_child;
+  } else if (_right == old_child) {
+    _right = new_child;
+  }
+}
+
+template <typename K, typename V, typename COMPARATOR, typename ALLOCATOR>
+inline typename RBTree<K, V, COMPARATOR, ALLOCATOR>::RBNode*
+RBTree<K, V, COMPARATOR, ALLOCATOR>::RBNode::rotate_left() {
+  // Move node down to the left, and right child up
+  RBNode* old_right = _right;
+
+  _right = old_right->_left;
+  if (old_right->_left != nullptr) {
+    old_right->_left->_parent = this;
+  }
```

Comment:
```suggestion
  if (_right != nullptr) {
    _right->_parent = this;
  }
```
Alternatively  a `set_parent(node)`, equivalent to the existing `set_child()`, with nullptr handling, would be even better